packages/grafana-ui/src/components/Table/ImageCell.tsx

```diff
@@ -1,14 +1,45 @@
-import React, { FC } from 'react';
+import { LinkModel } from '@grafana/data';
+import React, { FC, MouseEventHandler } from 'react';
 import { TableCellProps } from './types';
 
 export const ImageCell: FC<TableCellProps> = (props) => {
-  const { field, cell, tableStyles, cellProps } = props;
+  const { field, cell, tableStyles, row, cellProps } = props;
 
   const displayValue = field.display!(cell.value);
 
+  let link: LinkModel<any> | undefined;
+  let onClick: MouseEventHandler<HTMLAnchorElement> | undefined;
+
+  if (field.getLinks) {
+    link = field.getLinks({
+      valueRowIndex: row.index,
+    })[0];
+  }
+
+  if (link && link.onClick) {
+    onClick = (event) => {
+      // Allow opening in new tab
+      if (!(event.ctrlKey || event.metaKey || event.shiftKey) && link!.onClick) {
+        event.preventDefault();
+        link!.onClick(event);
+      }
+    };
+  }
+
```

Comment:
There seems to be some code duplication between ImageCell and DefaultCell can we extract a common utilty function/component?